modules/stub/src/stub.js

```diff
@@ -1,241 +1,282 @@
 ((win, doc) => {
 
+  const TCF_API_NAME = '__tcfapi';
+  const POST_MESSAGE_EVENT = 'message';
+  const TCF_POST_MSG_CALL = TCF_API_NAME + 'Call';
+  const TCF_POST_MSG_RETURN = TCF_API_NAME + 'Return';
+  const TCF_LOCATOR_NAME = TCF_API_NAME + 'Locator';
+  const CMP_STATUS = 'stub';
+  const TARGET_ORIGIN = '*';
+  const PING_COMMAND = 'ping';
+  const GDPR_APPLIES_COMMAND = 'setGdprApplies';
+
   const makeStub = () => {
 
-    const TCF_API_NAME = '__tcfapi';
-    const EVENT_NAME = 'message';
-    const TCF_POST_MSG_CALL = TCF_API_NAME + 'Call';
-    const TCF_POST_MSG_RETURN = TCF_API_NAME + 'Return';
-    const TCF_LOCATOR_NAME = TCF_API_NAME + 'Locator';
-    const CMP_STATUS = 'stub';
-    const TARGET_ORIGIN = '*';
-    const PING_COMMAND = 'ping';
-    const GDPR_APPLIES_COMMAND = 'setGdprApplies';
+    const isLeader = (win === win.top && win[TCF_API_NAME] === undefined);
+
+    const getPoxyingMethod = () => {
+
+      const callMap = new Map();
+      let callId = 0;
+
+      return (command, version, callback, parameter) => {
+
+        const call ={
+          callback,
+          command,
+          version,
+          parameter,
+        };
+
+        callMap.set(callId, call);
 
-    if (!(win.frames[TCF_LOCATOR_NAME])) {
+        win.top.postMessage({
```

Comment:
hmmm... Yeah so I think right now a locator frame exists anywhere there is a stub...  So it would have to walk to the top frame.

Reply:
that assumes there is a stub in the top frame.  Though unusual, it is possible and valid for the CMP to be loaded in an iframe instead of the top of the page.